src/api/services/adminServiceClient.ts

```diff
@@ -81,11 +81,38 @@ export const getSite = async (siteId: number): Promise<SiteDTO> => {
   return response.data;
 };
 
-export const getApiKeys = async (siteId: number): Promise<ApiKeyAdmin[]> => {
+export const getApiKeysFromAdmin = async (siteId: number): Promise<ApiKeyAdmin[]> => {
   const response = await adminServiceClient.get<ApiKeyAdmin[]>(`/api/client/list/${siteId}`);
   return response.data;
 };
 
+export const getApiKeyFromAdmin = async (keyId: String): Promise<ApiKeyAdmin | undefined> => {
+  const response = await adminServiceClient.get<ApiKeyAdmin>(`/api/client/keyId`, {
+    params: { keyId },
+  });
+
+  if (response.status === 404) {
```

Comment:
Can't really see a way we currently nicely deal with errors from admin. My idea here was a 404 means it can't find the key and that's the same message we'd want to return. Other errors are more likely something actually broken.